code_samples/back_office/suggestions/src/EventSubscriber/MySuggestionEventSubscriber.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+namespace App\EventSubscriber;
+
+use Ibexa\Contracts\ProductCatalog\ProductServiceInterface;
+use Ibexa\Contracts\ProductCatalog\Values\Product\ProductQuery;
+use Ibexa\Contracts\ProductCatalog\Values\Product\Query\Criterion;
+use Ibexa\Contracts\Search\Event\BuildSuggestionCollectionEvent;
+use Ibexa\Contracts\Search\Model\Suggestion\ContentSuggestion;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class MySuggestionEventSubscriber implements EventSubscriberInterface
+{
+    use LoggerAwareTrait;
+
+    private ProductServiceInterface $productService;
+
+    public function __construct(ProductServiceInterface $productService)
+    {
+        $this->productService = $productService;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            BuildSuggestionCollectionEvent::class => ['onBuildSuggestionCollectionEvent', -1],
+        ];
+    }
+
+    public function onBuildSuggestionCollectionEvent(BuildSuggestionCollectionEvent $event): BuildSuggestionCollectionEvent
+    {
+        $suggestionQuery = $event->getQuery();
+        $suggestionCollection = $event->getSuggestionCollection();
+
+        $text = $suggestionQuery->getQuery();
+        $words = explode(' ', preg_replace('/\s+/', ' ', $text));
+        $limit = $suggestionQuery->getLimit();
+
+        try {
+            $productQuery = new ProductQuery(null, new Criterion\LogicalOr([
+                new Criterion\ProductName("$text*"),
+                new Criterion\ProductCode($words),
+                new Criterion\ProductType($words),
+            ]), [], 0, $limit);
+            $searchResult = $this->productService->findProducts($productQuery);
+
+            if ($searchResult->getTotalCount()) {
+                $maxScore = 0.0;
+                /** @var \Ibexa\Contracts\Search\Model\Suggestion\ContentSuggestion $suggestion */
+                foreach ($suggestionCollection as $suggestion) {
+                    $maxScore = max($suggestion->getScore(), $maxScore);
```

Comment:
The variable ```$score``` determines the sequence of suggestions. Subsequently, based on this sequence, any items exceeding the specified limit are eliminated

Reply:
That's my point in https://github.com/ibexa/documentation-developer/pull/2222/files#diff-ec74d8d27848528a1597b553f88307e2bf87a1f66ee16a0cd6a70791b160ad4dR24. Matching products are always on top. If there is more than 5, only 5 product are there. If there is less than 5, some other results are displayed bellow. I'll reword this introduction to the example to be more clear.

Reply:
I have detailed the example against sorting and truncation in fbae6adf